api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubCommitV3.java

```diff
@@ -0,0 +1,318 @@
+            if (node != null) {
+                List<String> parentShas = getParentShas(node);
+                commit.setScmParentRevisionNumbers(parentShas);
+                commit.setFirstEverCommit(CollectionUtils.isEmpty(parentShas));
+                commit.setType(getCommitType(parentShas.size(), message, gitHubWebHookSettings, commitExclusionPatterns));
+
+                Object authorObject = node.get("author");
+                Object userObject = restClient.getAsObject(authorObject, "user");
+                String authorLogin = (userObject == null) ? "unknown" : restClient.getString(userObject, "login");
+                commit.setScmAuthorLogin(authorLogin);
+
+                if (StringUtils.isEmpty(authorLDAPDN)) {
+                    String authorLDAPDNFetched = StringUtils.isEmpty(authorLogin) ? null : getLDAPDN(repoUrl, authorLogin, gitHubWebHookToken);
+                    commit.setScmAuthorLDAPDN(authorLDAPDNFetched);
+                }
+            }
+
+            int numberChanges = 0;
+            if ((cObj.get("added") != null) && (cObj.get("added") instanceof List)) {
+                numberChanges += ((List) cObj.get("added")).size();
+            }
+            if ((cObj.get("removed") != null) && (cObj.get("removed") instanceof List)) {
+                numberChanges += ((List) cObj.get("removed")).size();
+            }
+            if ((cObj.get("modified") != null) && (cObj.get("modified") instanceof List)) {
+                numberChanges += ((List) cObj.get("modified")).size();
+            }
+
+            commit.setNumberOfChanges(numberChanges);
+
+            setCommitPullNumber(commit);
+
+            setCollectorItemId(commit, collectorItem);
+            commitsList.add(commit);
+        }
+
+        // For a merge commit for "Rebase and Merge"
+        setCommitPullNumbersForRebaseAndMergeCommit(commitsList);
+
+        return commitsList;
+    }
+
+    protected void setCommitPullNumbersForRebaseAndMergeCommit(List<Commit> commitsList) {
+
+        List<Commit> commitsWithPullNumber = Optional.ofNullable(commitsList)
+                                            .orElseGet(Collections::emptyList).stream()
+                                            .filter(commit -> !StringUtils.isEmpty(commit.getPullNumber()))
+                                            .collect(Collectors.toList());
+
+        // In case of Rebase and Merge, only the last commit in the list of commits on the "merge commit" json should have the PR number.
+        // This is because, there should be a corresponding PR in the DB whose "merge_commit_sha" matches the last commit in the list of commits
+        if (!CollectionUtils.isEmpty(commitsWithPullNumber)
+                && (commitsWithPullNumber.size() == 1)
+                && (commitsList.size() > 1)) {
+            Commit commitWithPR = commitsWithPullNumber.get(0);
+
+            Optional.ofNullable(commitsList)
```

Comment:
Optional.ofNullable may not be required as as you are checking for size() in line 204

Reply:
Good Catch, thanks. I have cleaned up the null checks before "instanceof" usage.

Reply:
Have removed the Optional usage.